mqlc/mqlc.go

```diff
@@ -1684,18 +1684,133 @@ func (c *compiler) postCompile() {
 			chunk, typ, ref := c.expandListResource(chunk, ref)
 			switch chunk.Id {
 			case "$one", "$all", "$none", "$any":
+				// default fields
 				ref = chunk.Function.Binding
-				chunk := code.Chunk(chunk.Function.Binding)
+				chunk := code.Chunk(ref)
 				typ = types.Type(chunk.Function.Type)
-				c.expandResourceFields(chunk, typ, ref)
-				block.Datapoints = append(block.Datapoints, block.TailRef(ref))
+				expanded := c.expandResourceFields(chunk, typ, ref)
+				// when no defaults are definied or query isn't about a resource, no block was added
+				if expanded {
+					block.Datapoints = append(block.Datapoints, block.TailRef(ref))
+					c.addValueFieldChunks(ref)
+				}
 			default:
 				c.expandResourceFields(chunk, typ, ref)
 			}
 		}
 	}
 }
 
+// addValueFieldChunks takes the value fields of the assessment and adds them to the
+// block for the default fields
+// This way, the actual data of the assessment automatically shows up in the output
+// of the assessment that failed the assessment
+func (c *compiler) addValueFieldChunks(ref uint64) {
+	var whereChunk *llx.Chunk
+
+	// find chunk with where/whereNot function
+	// it holds the reference to the block with the predicate(s) for the assessment
+	for {
+		chunk := c.Result.CodeV2.Chunk(ref)
+		if chunk.Function == nil {
+			// this is a safe guard and shouldn't happen
+			log.Error().Msg("failed to find where function for assessment")
+			return
+		}
+		if chunk.Function.Args != nil {
```

Comment:
I'm curious to better understand why we look for the Args in the chunk, instead of looking for either of the methods where/whereNot?

Reply:
Good point. There is no reason :) I made it more complicated than it needed to be. Fixed this now.